Subject: [PATCH] Init MemoryIntArray#mFd with -1

Symptom:
mFd was left uninitialized when constructor was failed.
As a result, it tried to close mFd=0 at finalize and
it crashed with SIGABRT.

diff --git a/core/java/android/util/MemoryIntArray.java b/core/java/android/util/MemoryIntArray.java
index 83e693c..6bf9b6d 100644
--- a/core/java/android/util/MemoryIntArray.java
+++ b/core/java/android/util/MemoryIntArray.java
@@ -53,7 +53,7 @@
 
     private final boolean mIsOwner;
     private final long mMemoryAddr;
-    private int mFd;
+    private int mFd = -1;
 
     /**
      * Creates a new instance.
